app/controllers/agents/rdvs_controller.rb

```diff
@@ -25,7 +25,12 @@ def edit
 
   def update
     authorize(@rdv)
-    flash[:notice] = 'Le rendez-vous a été modifié.' if @rdv.update(rdv_params)
+    if params[:status] == 'cancel' && @rdv.update(status: :excused, cancelled_at: Time.now)
```

Comment:
why doesn't the form send params[:status] = 'excused' ? it feels weird to 'swap' it manually here. 

also, I'm really surprised the status is named 'excused' and not `canceled`. Even more so since you're setting `cancelled_at` simultaneously.

Reply:
the "Annuler" button is not a form, just a button with ~~status~~ event params set to cancel.